iterative/utils/analytics.go

```diff
@@ -0,0 +1,290 @@
+		t0, _ := time.Parse(layout, matches[0])
+		taskDuration = t1.Sub(t0).Seconds()
+	}
+
+	return taskDuration
+}
+
+func IsCI() bool {
+	if _, ok := os.LookupEnv("CI"); ok {
+		return true
+	}
+
+	return false
+}
+
+func guessCI() string {
+	if _, ok := os.LookupEnv("GITHUB_SERVER_URL"); ok {
+		return "github"
+	}
+
+	if _, ok := os.LookupEnv("CI_SERVER_URL"); ok {
+		return "gitlab"
+	}
+
+	if _, ok := os.LookupEnv("BITBUCKET_WORKSPACE"); ok {
+		return "bitbucket"
+	}
+
+	return ""
+}
+
+func TerraformVersion() string {
+	var out bytes.Buffer
+	cmd := exec.Command("terraform", "--version")
+	cmd.Stdout = &out
+
+	err := cmd.Run()
+
+	if err == nil {
+		regex := regexp.MustCompile(`v\d+\.\d+\.\d+`)
+		version := regex.FindString(out.String())
+		return version
+	} else {
+		logrus.Error("Analytics: Failed extracting terraform version")
+	}
+
+	return ""
+}
+
+func GroupId() string {
+	if !IsCI() {
+		return ""
+	}
+
+	id, err := deterministic(fmt.Sprintf("%s%s%s%s%s",
+		os.Getenv("GITHUB_SERVER_URL"),
+		os.Getenv("GITHUB_REPOSITORY_OWNER"),
+		os.Getenv("CI_SERVER_URL"),
+		os.Getenv("CI_PROJECT_ROOT_NAMESPACE"),
+		os.Getenv("BITBUCKET_WORKSPACE"),
+	))
+
+	if err != nil {
+		return ""
+	}
+
+	return id.String()
+}
+
+func UserId() string {
+	id := uuid.New().String()
+	old := appdirs.UserConfigDir("dvc/user_id", "iterative", "", false)
+	new := appdirs.UserConfigDir("telemetry", "iterative", "", false)
+
+	_, errorOld := os.Stat(old)
+	if !os.IsNotExist(errorOld) {
+		os.Rename(old, new)
+	}
+
+	_, errorNew := os.Stat(new)
+	if !os.IsNotExist(errorNew) {
+		dat, _ := ioutil.ReadFile(new)
+		id = string(dat[:])
+	} else {
+		ioutil.WriteFile(new, []byte(id), 0644)
+	}
+
+	if IsCI() {
+		uid, err := deterministic(fmt.Sprintf("%s%s%s",
+			os.Getenv("GITHUB_ACTOR"),
+			os.Getenv("GITLAB_USER_ID"),
+			os.Getenv("BITBUCKET_STEP_TRIGGERER_UUID"),
+		))
+
+		if err == nil {
+			id = uid.String()
+		}
+	}
+
+	return id
+}
+
+func ResourceData(d *schema.ResourceData) map[string]interface{} {
+	if d == nil {
+		return map[string]interface{}{}
+	}
+
+	tpiLogs := d.Get("logs").([]interface{})
+	logs := ""
+	for _, log := range tpiLogs {
+		logs += log.(string)
+	}
+	spot := d.Get("spot").(float64)
+	return map[string]interface{}{
+		"cloud":           d.Get("cloud").(string),
+		"cloud_region":    d.Get("region").(string),
+		"cloud_machine":   d.Get("machine").(string),
+		"cloud_disk_size": d.Get("disk_size").(int),
+		"cloud_spot":      spot,
+		"cloud_spot_auto": spot == 0.0,
+		"task_status":     d.Get("status").(map[string]interface{}),
+		"task_duration":   TaskDuration(logs),
+		"task_resumed":    len(tpiLogs) > 1,
+	}
+}
+
+func JitsuEventPayload(action string, e error, extra map[string]interface{}) map[string]interface{} {
+	systemInfo := SystemInfo()
+
+	extra["ci"] = guessCI()
+	extra["terraform_version"] = TerraformVersion()
+
+	err := ""
+	if e != nil {
+		err = reflect.TypeOf(e).String()
```

Comment:
### Continuation of https://github.com/iterative/terraform-provider-iterative/pull/493#discussion_r858526582

If we want to provide more insightful information, we can either send the whole logs (handle with care) or try to extract anonymized information from error messages; for example, match some strings.

Reply:
It is a big effort and, yes, it will require regular expressions. It would involve collecting as many errors as possible (from our own runs) and then finding out what we can send without including sensitive information.

Sounds like a good follow-up pull request once I feel really bored. 😉

Reply:
so we are ok right now?

Reply:
Yes, this should probably be a follow-up issue, but not something to be addressed immediately.